lib/matplotlib/pyplot.py

```diff
@@ -543,9 +531,26 @@ def make_active(event):
         _pylab_helpers.Gcf.set_active(figManager)
         figManager.canvas.figure.number = num
 
+        if _INSTALL_FIG_OBSERVER:
+            figManager.canvas.figure.add_callback(_auto_draw_if_interactive)
```

Comment:
Why limit this to figures created by pyplot?  As far as I see it this takes us quite some distance towards implementing MEP26 (Dynamic Artist Style Sheets).